docs/concepts/models.md

```diff
@@ -1419,99 +1469,11 @@ print(error_locations)
 #> [('a',), ('b',), ('c',), ('d',), ('e',)]
 ```
 
-## Required fields
-
-To declare a field as required, you may declare it using an annotation, or an annotation in combination with a
-[`Field`][pydantic.Field] function (without specifying any `default` or `default_factory` argument).
-
-```python
-from pydantic import BaseModel, Field
-
-
-class Model(BaseModel):
-    a: int
-    b: int = Field(alias='B')
-    c: int = Field(..., alias='C')
-```
-
-Here `a`, `b` and `c` are all required. The field `c` uses the [ellipsis][Ellipsis] as a default argument,
-emphasizing on the fact that it is required. However, the usage of the [ellipsis][Ellipsis] is discouraged
-as it doesn't play well with type checkers.
-
-!!! note
-    In Pydantic V1, fields annotated with `Optional` or `Any` would be given an implicit default of `None` even if no
-    default was explicitly specified. This behavior has changed in Pydantic V2, and there are no longer any type
-    annotations that will result in a field having an implicit default value.
-
-    See [the migration guide](../migration.md#required-optional-and-nullable-fields) for more details on changes
-    to required and nullable fields.
-
-## Fields with non-hashable default values
-
-A common source of bugs in python is to use a mutable object as a default value for a function or method argument,
-as the same instance ends up being reused in each call.
-
-The `dataclasses` module actually raises an error in this case, indicating that you should use the `default_factory`
-argument to `dataclasses.field`.
-
-Pydantic also supports the use of a [`default_factory`](#fields-with-dynamic-default-values) for non-hashable default
-values, but it is not required. In the event that the default value is not hashable, Pydantic will deepcopy the default
-value when creating each instance of the model:
-
-```python
-from typing import Dict, List
-
-from pydantic import BaseModel
-
-
-class Model(BaseModel):
-    item_counts: List[Dict[str, int]] = [{}]
-
-
-m1 = Model()
-m1.item_counts[0]['a'] = 1
-print(m1.item_counts)
-#> [{'a': 1}]
-
-m2 = Model()
-print(m2.item_counts)
-#> [{}]
-```
-
-## Fields with dynamic default values
-
-When declaring a field with a default value, you may want it to be dynamic (i.e. different for each model).
-To do this, you may want to use a `default_factory`.
-
-Here is an example:
-
-```python
-from datetime import datetime, timezone
-from uuid import UUID, uuid4
-
-from pydantic import BaseModel, Field
-
-
-def datetime_now() -> datetime:
-    return datetime.now(timezone.utc)
-
-
-class Model(BaseModel):
-    uid: UUID = Field(default_factory=uuid4)
-    updated: datetime = Field(default_factory=datetime_now)
-
-
-m1 = Model()
-m2 = Model()
-assert m1.uid != m2.uid
-```
```

Comment:
Why not include this example on the fields page? Can be in a collapsable part, if you want...

Reply:
The fields page already has an example using a `default_factory`:

```python
class User(BaseModel):
    id: str = Field(default_factory=lambda: uuid4().hex)
```